src/ethereum/osaka/vm/instructions/system.py

```diff
@@ -742,3 +743,52 @@ def revert(evm: Evm) -> None:
 
     # PROGRAM COUNTER
     pass
+
+
+def pay(evm: Evm) -> None:
+    """
+    Transfer ether to an account.
+
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+    """
+    # STACK
+    to = to_address_without_mask(pop(evm.stack))
+    value = pop(evm.stack)
+
+    # GAS
+    if to in evm.accessed_addresses:
+        access_gas_cost = GAS_WARM_ACCESS
+    else:
+        evm.accessed_addresses.add(to)
+        access_gas_cost = GAS_COLD_ACCOUNT_ACCESS
+
+    create_gas_cost = (
+        Uint(0)
+        if is_account_alive(evm.message.block_env.state, to) or value == 0
+        else GAS_NEW_ACCOUNT
+    )
+
+    transfer_gas_cost = Uint(0) if value == U256(0) else GAS_CALL_VALUE
+
+    charge_gas(evm, access_gas_cost + create_gas_cost + transfer_gas_cost)
+
+    # OPERATION
+    if evm.message.is_static:
+        raise WriteInStaticContext("Cannot PAY in static context")
```

Comment:
In the spec, this check is done before any other operation (so also before popping any value from stack). I don't think moving this around in this method would yield a different outcome, but it would match the spec better to do this check prior to anything else. So I would suggest moving this check up.

Reply:
Yes. This does not make a difference to the outcome since all the gas in consumed and warmed addresses are discarded upon error. However, I implemented it in this order since that makes it consistent with the implementation of `CALL*` and other methods where you check if the frame is static.